slsDetectorServers/slsDetectorServer/src/nios.c

```diff
@@ -126,4 +129,50 @@ int mapCSP0(void) {
     return OK;
 }
 
-u_int32_t *Nios_getBaseAddress() { return csp0base; }
\ No newline at end of file
+u_int32_t *Nios_getBaseAddress() { return csp0base; }
+
+int Nios_checkKernelVersion(char *expectedVersion) {
+    // extract kernel date string
+    struct utsname buf;
+    if (uname(&buf) == -1) {
+        LOG(logERROR, ("Could not get kernel version\n"));
+        return FAIL;
+    }
+
+    // remove first word (#version number)
+    const char *ptr = strchr(buf.version, ' ');
```

Comment:
Are we sure that this search always works? On my machine it's the second whitespace + 1.

Reply:
Yes, exactly. That's why i put this into Nios.c. Because in the actual nios board, it doesnt have the SMP,  just one word before.